### DISCUSSION TITLE
Failed to convert Message

### DISCUSSION THREAD

**Discussion by Vinniebonjour** [gray-learn/collectorspot#2] [Q&A] [answered]
How do you fix “Failed to convert Serialization Message content”

**Comment by gray-learn** [comment, marked as answer]
Once you Failed to convert Serialization Message content.
In MessagingConfig, you can use Jackson2JsonMessageConverter and  ObjectMapper
A. Publisher // Serialization

@Configuration
@EnableRabbit
public class MessagingConfig implements InitializingBean {
    // JSON
    @Resource
    private RabbitTemplate rabbitTemplate;

    @Override
    public void afterPropertiesSet() {
        // Serialize the JSON
        rabbitTemplate.setMessageConverter(new Jackson2JsonMessageConverter());
    }
    // JSON
}
B. Consumer // Deserialization

@Configuration
@EnableRabbit
public class MessagingConfig {

    @Bean
    public MessageConverter jsonMessageConverter(ObjectMapper objectMapper){
        return new Jackson2JsonMessageConverter(objectMapper);
    }
}

**Comment by Vinniebonjour** [comment]
thanks for solutions
